h2o-core/src/main/java/water/H2ONode.java

```diff
@@ -146,32 +147,57 @@ public boolean isHealthy(long now) {
   static private final AtomicInteger UNIQUE = new AtomicInteger(1);
   static H2ONode IDX[] = new H2ONode[1];
 
+  static H2ONode[] getClients(){
+    ArrayList<H2ONode> clients = new ArrayList<>();
+    for( Map.Entry<H2Okey, H2ONode> entry : INTERN.entrySet()){
+      if (entry.getValue()._heartbeat._client) {
+        clients.add(entry.getValue());
+      }
+    }
+    return clients.toArray(new H2ONode[0]);
+  }
+
+  static H2ONode getClientByIPPort(String ipPort){
+    for( Map.Entry<H2Okey, H2ONode> entry : INTERN.entrySet()){
+      if (entry.getValue()._heartbeat._client && entry.getValue().getIpPortString().equals(ipPort)) {
+        return entry.getValue();
+      }
+    }
+    return null;
+  }
+
+  static boolean removeClient(H2ONode node){
+      // Before we remove the Client, stop the sending thread
+      node.stopSendThread();
+      Log.info("Removing client: " + node.toDebugString());
+      return INTERN.remove(node._key, node);
```

Comment:
for consistency it might be better to first remove from INTERN and then stop the thread; it would be nice to have an invariant that says that each node in intern map does have a send thread

Reply:
Agree, thanks!